rust/blockstore/src/arrow/provider.rs

```diff
@@ -61,47 +62,53 @@ impl ArrowBlockfileProvider {
         }
     }
 
-    pub fn create<
+    pub async fn get_writer<
```

Comment:
nit to self - need better name

Reply:
do you have any suggestions?

Reply:
Maybe we change open -> read() and this to write() with fork as an option.

Reply:
sure, I like that

Reply:
I renamed `get_writer()` to `write()`, will rename `open() -> read()` in a future PR and update `create()`/`fork()`usage